### DISCUSSION TITLE
Persist changed language without NEXT_LOCALE cookie

### DISCUSSION THREAD

**Discussion by hartherbert** [vinissimus/next-translate#506] [Q&A]
My nextjs app is running in a third-party context (iframe) and now with all the new cookie regulations (no third-party cookies anymore), it is not possible for me to [set the NEXT_LOCALE cookie](https://github.com/vinissimus/next-translate#10-how-to-save-the-user-defined-language).

I was wondering if any of you knows a way to force nextjs to use a specified locale without the cookie? Maybe localStorage?

Thanks for any response!

**Comment by aralroca** [comment]
You can use the localStorage and create a custom hook to redirect to the preferred locale, but then the redirect will be done in client-side...
